packages/synapse-react-client/test/lib/containers/SynapseNavDrawer.test.tsx

```diff
@@ -1,41 +1,160 @@
-import { render, screen } from '@testing-library/react'
+import { render, screen, within } from '@testing-library/react'
 import React from 'react'
 import {
   SynapseNavDrawer,
   SynapseNavDrawerProps,
 } from '../../../src/lib/containers/SynapseNavDrawer'
 import { createWrapper } from '../../testutils/TestingLibraryUtils'
 import { SynapseContextType } from '../../../src/lib/utils/SynapseContext'
-import { server } from '../../../mocks/msw/server'
-import { INVALID_ACCESS_TOKEN_CAUSES_GET_PROFILE_ERROR } from '../../../mocks/user/mock_user_profile'
+import {
+  MOCK_USER_ID,
+  MOCK_USER_ID_2,
+  mockUserBundle,
+} from '../../../mocks/user/mock_user_profile'
+import { SynapseClient } from '../../../src/lib'
+import { SubmissionState } from '../../../src/lib/utils/synapseTypes'
+import { SubmissionSearchResult } from '../../../src/lib/utils/synapseTypes/AccessSubmission'
+import { mockManagedACTAccessRequirement } from '../../../mocks/mockAccessRequirements'
+import { mockSubmittedSubmission } from '../../../mocks/dataaccess/MockSubmission'
 
 const defaultProps: SynapseNavDrawerProps = {
   initIsOpen: false,
   signoutCallback: jest.fn(),
 }
 
-function renderComponent(wrapperProps?: SynapseContextType) {
+function renderComponent(wrapperProps?: Partial<SynapseContextType>) {
   render(<SynapseNavDrawer {...defaultProps} />, {
     wrapper: createWrapper(wrapperProps),
   })
 }
 
+const numFilesInDownloadList = 10
+
+jest.spyOn(SynapseClient, 'getDownloadListStatistics').mockResolvedValue({
+  concreteType:
+    'org.sagebionetworks.repo.model.download.FilesStatisticsResponse',
+  totalNumberOfFiles: numFilesInDownloadList,
+  numberOfFilesAvailableForDownload: 5,
+  numberOfFilesAvailableForDownloadAndEligibleForPackaging: 4,
+  numberOfFilesRequiringAction: 5,
+  sumOfFileSizesAvailableForDownload: 1561981,
+})
+
+const openSubmissions: SubmissionSearchResult[] = [
+  {
+    id: mockSubmittedSubmission.id,
+    createdOn: mockSubmittedSubmission.submittedOn,
+    modifiedOn: mockSubmittedSubmission.submittedOn,
+    accessRequirementId: `${mockManagedACTAccessRequirement.id}`,
+    accessRequirementVersion: `${mockManagedACTAccessRequirement.versionNumber}`,
+    accessRequirementName: mockManagedACTAccessRequirement.name,
+    accessRequirementReviewerIds: [MOCK_USER_ID.toString()],
+    submitterId: MOCK_USER_ID_2.toString(),
+    accessorChanges: [],
+    state: SubmissionState.SUBMITTED,
+  },
+]
+
+jest.spyOn(SynapseClient, 'searchAccessSubmission').mockResolvedValue({
+  results: openSubmissions,
+})
+
+const mockGetUserBundle = jest.spyOn(SynapseClient, 'getMyUserBundle')
+
 describe('SynapseNavDrawer tests', () => {
-  beforeAll(() => server.listen())
-  afterEach(() => server.restoreHandlers())
-  afterAll(() => server.close())
```

Comment:
Drop mock-service-worker in these tests because it's a bit more challenging to manage different user states compared to just mocking SynapseClient